### ISSUE TITLE
bigquery udf: len is undefined

### ISSUE THREAD

**Issue by tsdlovell** [ibis-project/ibis#1470]
workaround is to use javascript notation in the udf

```
len = lambda x: x.length
```

which will transpile to

```
let len = ((x) => x.length)
```

**Comment by tsdlovell** [comment]
same is true of `zip` and probably a lot of other builtins, so maybe [this phrasing](http://docs.ibis-project.org/udf.html#bigquery) should change or be caveated

>  Most of the Python language is supported including classes, functions and generators

**Comment by cpcloud** [comment]
Most of the *language* is supported, builtin functions are not part of the language, they are part of the standard library. I'll make that distinction more clear there.

**Comment by cpcloud** [comment]
Pull requests always welcome to add builtins

**Comment by cpcloud** [comment]
@tsdlovell Are there any particular builtins you'd like to see other `len`? Happy to implement any high priority ones alongside `len`.

**Comment by tsdlovell** [comment]
For enumerate
```
enumerate = lambda x: x.map(lambda e, i: (e, i))
```
transpiles to
```
let enumerate = ((x) => x.map(((e, i) => [e, i])));
```

**Comment by cpcloud** [comment]
`enumerate` yields `index, element` you probably want to write that as `lambda x: x.map(lambda e, i: (i, e))`

**Comment by cpcloud** [comment]
If you are feeling adventurous you can add the rewrites yourself without having to wait on me to implement them. See here for some examples: https://github.com/ibis-project/ibis/blob/master/ibis/bigquery/udf/core.py#L83-L117

**Comment by tsdlovell** [comment]
@cpcloud , can you point me to any snippets where a more complicated statement is substituted?

**Comment by tsdlovell** [comment]
I might be able to start from [visit_ListComp](https://github.com/ibis-project/ibis/blob/master/ibis/bigquery/udf/core.py#L507)
